LearningMissionSimulation/PlaygroundGarush.cs

```diff
@@ -86,66 +116,170 @@ List<Module> GetModuleList()
+            ReportFooter(actionName: action);
         }
 
         public void CreateSubjects(int subjectCount)
         {
+            string action = "create subjects";
+            ReportHeader(actionName: action);
             int i = 0;
             while (i < subjectCount)
             {
+
                 Subject subject = ObjectGenerator.GenerateSubject();
                 subjectList.Add(subject);
                 subjectIdList.Add(subject.Id);
-                Console.WriteLine("\ncreate subject {0}\n", subject);
-                i++;
+
+                ReportItem(itemName: subject.ToString(), actionName: action, itemIndex: ++i);
             }
+
+            ReportSummary(actionName: action, count: subjectCount);
+            ReportFooter(actionName: action);
         }
 
         public void CreateModules(int moduleCount)
         {
+            string action = "create modules";
+            ReportHeader(actionName: action);
             int i = 0;
-            while (i < moduleCount)
+            if (subjectList.Count == 0)
+            {
+                ReportError(missingResource: "Subject", failedAction: action);
+            }
+            else
             {
-                Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count)];
-                Module module = ObjectGenerator.GenerateModule(subjectId);
-                moduleDictionary.Add(module.Id, module);
-                moduleIdList.Add(module.Id);
-                Console.WriteLine("\ncreate module {0}\n", module);
-                i++;
+                while (i < moduleCount)
+                {
+                    Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count)];
+                    Module module = ObjectGenerator.GenerateModule(subjectId);
+                    moduleDictionary.Add(module.Id, module);
+                    moduleIdList.Add(module.Id);
+
+                    ReportItem(itemName: module.ToString(), actionName: action, itemIndex: ++i);
+                }
             }
+
+            ReportSummary(actionName: action, count: moduleCount);
+            ReportFooter(actionName: action);
         }
 
         public void CreateClassrooms(int classroomCount)
         {
-            throw new NotImplementedException();
+            // throw new NotImplementedException();
+            string action = "create Classrooms";
+            int i = 0;
+            ReportHeader(actionName: action);
+
+            if (moduleIdList.Count == 0)
+            {
+                ReportError(missingResource: "Module", failedAction: action);
+            }
+            else
+            {
+                while (i < classroomCount)
+                {
+                    Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+                    Module module;
+                    moduleDictionary.TryGetValue(moduleId, out module);
```

Comment:
You have to check the return value of moduleDictionary.TryGetValue(moduleId, out module) -- before adding the module to the classroom.